packages/shared-base/src/LegacySettings/settings.ts

```diff
@@ -29,6 +29,21 @@ export const pluginIDsSettings = createGlobalSettings<Record<EnhanceableSite | E
     isEqual,
 )
 
+export const InjectSwitchSettings = createGlobalSettings<Record<EnhanceableSite, boolean>>(
```

Comment:
You should use `createBulkSettings`.

Reply:
Only the specified site has setting, key must be `EnhanceableSite`

Reply:
unfortunately, all other usages of `createBulkSettings` also limits key to `EnhanceableSite`. I hope we can keep consistent here.

Reply:
not resolved, plz change!